src/lib/linux-api/src/signal.rs

```diff
@@ -0,0 +1,1192 @@
+                        l_uid: child_uid,
+                        l_status: signal.into(),
+                        l_utime: child_utime,
+                        l_stime: child_stime,
+                    },
+                },
+            )
+        }
+    }
+
+    pub fn new_for_sigchld_killed(
+        child_pid: i32,
+        child_uid: u32,
+        fatal_signal: Signal,
+        child_utime: i64,
+        child_stime: i64,
+    ) -> Self {
+        Self::new_for_sigchld_signaled(
+            SigInfoCodeCld::CLD_KILLED,
+            child_pid,
+            child_uid,
+            fatal_signal,
+            child_utime,
+            child_stime,
+        )
+    }
+    pub fn new_for_sigchld_dumped(
+        child_pid: i32,
+        child_uid: u32,
+        fatal_signal: Signal,
+        child_utime: i64,
+        child_stime: i64,
+    ) -> Self {
+        Self::new_for_sigchld_signaled(
+            SigInfoCodeCld::CLD_DUMPED,
+            child_pid,
+            child_uid,
+            fatal_signal,
+            child_utime,
+            child_stime,
+        )
+    }
+    pub fn new_for_sigchld_trapped(
+        child_pid: i32,
+        child_uid: u32,
+        child_utime: i64,
+        child_stime: i64,
+    ) -> Self {
+        Self::new_for_sigchld_signaled(
+            SigInfoCodeCld::CLD_TRAPPED,
+            child_pid,
+            child_uid,
+            Signal::SIGTRAP,
+            child_utime,
+            child_stime,
+        )
+    }
+    pub fn new_for_sigchld_stopped(
+        child_pid: i32,
+        child_uid: u32,
+        child_utime: i64,
+        child_stime: i64,
+    ) -> Self {
+        Self::new_for_sigchld_signaled(
+            SigInfoCodeCld::CLD_STOPPED,
+            child_pid,
+            child_uid,
+            Signal::SIGSTOP,
+            child_utime,
+            child_stime,
+        )
+    }
+    pub fn new_for_sigchld_continued(
+        child_pid: i32,
+        child_uid: u32,
+        child_utime: i64,
+        child_stime: i64,
+    ) -> Self {
+        Self::new_for_sigchld_signaled(
+            SigInfoCodeCld::CLD_CONTINUED,
+            child_pid,
+            child_uid,
+            Signal::SIGCONT,
+            child_utime,
+            child_stime,
+        )
+    }
+
+    // TODO: (see sigaction(2))
+    // * new_for_sigill
+    // * new_for_sigfpe
+    // * new_for_sigsegv
+    // * new_for_sigtrap
+    // * new_for_poll
+    // * new_for_seccomp
+    // ...
+}
+
+impl Default for SigInfo {
+    fn default() -> Self {
+        unsafe { core::mem::zeroed() }
+    }
+}
+
+#[allow(non_camel_case_types)]
+pub type linux_sigset_t = bindings::linux_sigset_t;
+
+/// Compatible with the Linux kernel's definition of sigset_t on x86_64.
+///
+/// This is analagous to, but typically smaller than, libc's sigset_t.
+#[repr(transparent)]
+#[derive(Copy, Clone, Eq, PartialEq, Debug, Default, VirtualAddressSpaceIndependent)]
+pub struct SigSet(linux_sigset_t);
```

Comment:
nit: maybe make it `pub linux_sigset_t` so it can be accessed directly instead of through `peel_ref`?